### PR TITLE
Merge arrow_simple_demo into arrow_guide.

### PR DESCRIPTION
**Pull request by anntzer**
plt.arrow() (aka patches.FancyArrow) has a rather awkward API (#12768) and
doesn't warrant its own demo, but can be mentioned at the end of the
arrow guide together with patches.Arrow, which is nearly as awkward...

## PR Checklist

- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [ ] New features are documented, with examples if plot related.
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

**Review by story645 — APPROVED** [review]

File: examples/shapes_and_collections/arrow_guide.py [hunk 1]
```diff
@@ -81,28 +83,41 @@
...
 
 arrow = mpatches.Arrow(x_tail, y_tail, dx, dy)
 axs[1].add_patch(arrow)
-axs[1].set_xlim(0, 2)
-axs[1].set_ylim(0, 2)
+arrow = mpatches.FancyArrow(
+    x_tail, y_tail - .4, dx, dy, width=.1, length_includes_head=True)
+axs[1].add_patch(arrow)
+axs[1].arrow(
+    x_tail + 1, y_tail - .4, dx, dy, width=.1, length_includes_head=True)
```
Comment (story645) [inline examples/shapes_and_collections/arrow_guide.py:119]: I very much dig this example of showing how the patch is identical to the method.

**Review by story645 — APPROVED** [review]

File: examples/shapes_and_collections/arrow_guide.py [hunk 2]
```diff
@@ -81,28 +83,41 @@
...
+axs[1].set(xlim=(0, 2), ylim=(0, 2))
```
Comment (QuLogic) [inline examples/shapes_and_collections/arrow_guide.py:120]: Should it be ymax=2? Seems to be a lot of whitespace in the bottom Axes compared to the top one.

Comment (anntzer) [reply examples/shapes_and_collections/arrow_guide.py:120]: That's intentional to be consistent with the other figures created by this example and to show the distortion that occurs when drawing arrows in data coords.

File: examples/shapes_and_collections/arrow_guide.py [hunk 3]
```diff
@@ -81,28 +83,41 @@
...
 
 fig, axs = plt.subplots(nrows=2)
 
 arrow = mpatches.Arrow(x_tail, y_tail, dx, dy)
 axs[0].add_patch(arrow)
+arrow = mpatches.FancyArrow(
+    x_tail, y_tail - .4, dx, dy, width=.1, length_includes_head=True)
+axs[0].add_patch(arrow)
+axs[0].arrow(
+    x_tail + 1, y_tail - .4, dx, dy, width=.1, length_includes_head=True)
```
Comment (timhoffm) [inline examples/shapes_and_collections/arrow_guide.py:111]: Maybe color the arrows differently, so that it's easier to see which code created which arrow.

Comment (anntzer) [reply examples/shapes_and_collections/arrow_guide.py:111]: sure

File: examples/shapes_and_collections/arrow_guide.py [hunk 4]
```diff
@@ -81,28 +83,42 @@
...
 
 fig, axs = plt.subplots(nrows=2)
 
 arrow = mpatches.Arrow(x_tail, y_tail, dx, dy)
 axs[0].add_patch(arrow)
+arrow = mpatches.FancyArrow(x_tail, y_tail - .4, dx, dy,
+                            width=.1, length_includes_head=True, color="C1")
+axs[0].add_patch(arrow)
+axs[0].arrow(x_tail + 1, y_tail - .4, dx, dy,
+             width=.1, length_includes_head=True, color="C1")
```
Comment (timhoffm) [inline examples/shapes_and_collections/arrow_guide.py:112]:
```suggestion
             width=.1, length_includes_head=True, color="C2")
```

File: examples/shapes_and_collections/arrow_guide.py [hunk 5]
```diff
@@ -81,28 +83,42 @@
...
 
 arrow = mpatches.Arrow(x_tail, y_tail, dx, dy)
 axs[1].add_patch(arrow)
-axs[1].set_xlim(0, 2)
-axs[1].set_ylim(0, 2)
+arrow = mpatches.FancyArrow(x_tail, y_tail - .4, dx, dy,
+                            width=.1, length_includes_head=True, color="C1")
+axs[1].add_patch(arrow)
+axs[1].arrow(x_tail + 1, y_tail - .4, dx, dy,
+             width=.1, length_includes_head=True, color="C1")
```
Comment (timhoffm) [inline examples/shapes_and_collections/arrow_guide.py:120]:
```suggestion
             width=.1, length_includes_head=True, color="C2")
```

**Review by timhoffm — COMMENTED** [review]
It's better if all have different colors.

**Comment by anntzer** [comment]
sure

**Review by timhoffm — APPROVED** [review]
Anybody can merge after CI pass.

**Pull request merged**